### DISCUSSION TITLE
Viability of Filter::wildcard

### DISCUSSION THREAD

**Discussion by usernotnull** [worksome/envy#12] [General]
I made a github action to run envy as part of the pipeline.

When running composer without dev deps on the next job with `composer install --no-dev --no-interaction --prefer-dist`, using Filter::wildcard inside the config file will cause the below error, since the config file will still be processed even if envy is not required in the no-dev env:

```
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

In envy.php line 62:
                                                          
  Class "Worksome\Envy\Support\Filters\Filter" not found  
                                                          

Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1
Error: Process completed with exit code 1.
```

I could add envy not as a dev dependency, but it's not the ideal solution. At the same time, if I just remove the wildcard support, we go back to issue #3 .

Filter::wildcard is great but forces us to only use it without no-dev.

Thoughts?

**Comment by olivernybroe** [comment]
Hmm interesting issue. 

A solution that comes to mind is adding a method to the Filter interface called `applies` and takes the string value. 

This allows us to write a string called `wildcard:FOO_BAR_*` where the applies method checks for if the string starts with `wildcard`. 

For the equal filter it just checks if there is a string.
Might need to figure something out so equal filter is not applied when trying to use the wildcard filter 🤔 

WDYT?
